installation/README.md

```diff
@@ -1,9 +1,98 @@
+First, make sure that your pi-puck and computer are connected to the same local network.
+You can ping the pi-puck using:
+```
+ping raspberrypi.local
+```
+or if your device doesn't support `mDNS` you will need to use `nmap` to [discover pi-puck's IP address](https://www.raspberrypi.org/documentation/remote-access/ip-address.md).
+
+Then, connect to the pi-puck via SSH:
+```
+ssh pi@raspberrypi.local
+```
+with `raspberry` as a default password.
+
+> More about SSH and Raspberry Pi can be found [here](https://www.raspberrypi.org/documentation/remote-access/ssh/).
+
+ROS2 installation is located in `/home/pi/ros2_ws`, so you can navigate inside and start `epuck_ros2` driver as:
+```bash
+source install/local_setup.bash
+ros2 launch webots_ros2_epuck2 robot_launch.py
+```
+
+## Fresh ROS2 Installation
+In certain use-cases, a user may prefer to install ROS2, `epuck_ros2`, configure the device tree and install dependencies manually.
+Even not recommended, this is the only way to install ROS2 version other than the ones provided by us.
+Please note that neither ARMv6 architecture, neither Debian (Raspberry Pi OS is based on Debian) are supported by ROS2 and therefore, you may need to manually solve the compilation issues.
+
+After the Raspberry Pi OS is booted and you have SSH connection please execute the following command:
 ```
 curl https://raw.githubusercontent.com/cyberbotics/epuck_ros2/master/installation/ros-install.sh | sh
-```
\ No newline at end of file
+```
+> Note that this script is in very early stage and it may fail.
+In case of failure try to understand which package has caused the error and check if the community on Github has the solution.
+Also, there are around 200 ROS2 packages and it will take a long time to compile.
+
+The script will install dependencies, download ROS2 locally, compile and configure the device tree necessary for Raspberry Pi to communicate with the e-puck2 base.
+
+### Creating Custom Image
+
+As the installation is over or you developed a custom node you may want to save the configuration and quickly distribute to the other robots.
+To achieve that you can simply put the SD card in your computer and copy the content to `*.img`:
+```bash
+dd if=/dev/sdX of=path_to_pi_puck_image.img
+```
+The image size will be equal (if expanded before) to your SD card size even though the content of the image is much smaller.
+To shrink the image use the following command:
+```bash
+curl https://raw.githubusercontent.com/Drewsif/PiShrink/master/pishrink.sh | sudo bash -s -- path_to_pi_puck_image.img
+```
+
+Now, burn the `path_to_pi_puck_image.img` to other SD cards that will be used in order pi-pucks.
+
+## Starting `epuck_ros2` on Boot
+In most of the cases, ROS2 development will take place on your computer and ROS2 will take handle the communication between the local nodes and nodes on the pi-puck.
+In that scenario, you may want to execute `epuck_ros2` as soon as pi-puck boots without manual intervention.
```

Comment:
Is this the case with the provided image or not?

Reply:
Currently, it is the case for both. Maybe we should start the driver on the boot in our image by default?

Reply:
It's probably simpler for the user indeed.